src/tests/InputProblem.test.tsx

```diff
@@ -0,0 +1,184 @@
+  await screen.findByText('Please provide an number')
+})
+
+test('InputProblem textbox is expecting Integer but input was text.', async () => {
+  render(
+          <InputProblem
+          solvedCallback={() => {}}
+          exhaustedCallback={() => {}}
+          allowedRetryCallback={() => {}}
+          content={'Content'}
+          correctResponse={''}
+          encourageResponse={''}
+          retryLimit={0}
+          solution={' 5 '}
+          buttonText={'Submit'}
+          comparator={'integer'}
+          />
+  )
+  await act(async () => {
+    fireEvent.change(screen.getByRole('textbox'), { target: { value: 'Wrong input!' } })
+    screen.getByRole('button').click()
+  })
+  await screen.findByText('Please provide an Integer')
+})
+
+test('InputProblem button click with wrong answer should evaluate to incorrect', async () => {
+  const solvedHandler = jest.fn()
+  const exhaustedHandler = jest.fn()
+  const allowedRetryHandler = jest.fn()
+
+  render(
+          <InputProblem
+          solvedCallback={solvedHandler}
+          exhaustedCallback={exhaustedHandler}
+          allowedRetryCallback={allowedRetryHandler}
+          content={'Content'}
+          correctResponse={''}
+          encourageResponse={'Try again!'}
+          retryLimit={0}
+          solution={' 5 '}
+          buttonText={'Submit'}
+          comparator={'integer'}
+          />
+  )
+  await act(async () => {
+    fireEvent.change(screen.getByRole('textbox'), { target: { value: '4' } })
+    screen.getByRole('button').click()
+  })
+  await screen.findByText('Try again!')
+  expect(solvedHandler).toBeCalledTimes(0)
+  expect(exhaustedHandler).toBeCalledTimes(0)
+  expect(allowedRetryHandler).toBeCalledTimes(1)
+})
+
+test('Retry limit, encourageResponse, and exausted callback test', async () => {
+  const solvedHandler = jest.fn()
+  const exhaustedHandler = jest.fn()
+  const allowedRetryHandler = jest.fn()
+
+  render(
+          <InputProblem
+          solvedCallback={solvedHandler}
+          exhaustedCallback={exhaustedHandler}
+          allowedRetryCallback={allowedRetryHandler}
+          content={'Content'}
+          correctResponse={'Correct!'}
+          encourageResponse={'Try again!'}
+          retryLimit={3}
+          solution={' 5 '}
+          buttonText={'Submit'}
+          comparator={'integer'}
+          />
+  )
+  await act(async () => {
+    fireEvent.change(screen.getByRole('textbox'), { target: { value: '4' } })
+    screen.getByRole('button').click()
+    screen.getByRole('button').click()
+    screen.getByRole('button').click()
+  })
+  await screen.findByText('Try again!')
+
+  await act(async () => {
+    fireEvent.change(screen.getByRole('textbox'), { target: { value: '3' } })
+    screen.getByRole('button').click()
+  })
+
+  await screen.findByText('No more attempts allowed')
+  expect(solvedHandler).toBeCalledTimes(0)
+  expect(exhaustedHandler).toBeCalledTimes(1)
+  expect(allowedRetryHandler).toBeCalledTimes(3)
```

Comment:
Can we add assertions that the form is disabled when the form is exhausted to this test case?